src/arr/base/contracts.arr

```diff
@@ -33,13 +40,17 @@ data FieldFailure:
   | missing-field(loc, field) with:
     render-reason(self, loc, from-fail-arg):
       [ED.error:
-        [ED.para: ED.text("Missing field"), ED.code(ED.text(self.field)),
-          ED.text("is required at"), draw-and-highlight(self.loc)]]
+        [ED.para: ED.text("Missing field "), ED.code(ED.text(self.field)),
```

Comment:
Given that going through each error message one-at-a-time is precisely what I have been doing for the past few months and will continue to do for the foreseeable future, fixes relating to missing spaces come essentially for free.

One change in my branch is I've been moving towards a more semantically meaningful `ErrorDisplay`. `ED.para` no longer expands to an `h-sequence` in the errors branch but rather an actual `paragraph` [variant](https://github.com/brownplt/pyret-lang/pull/696/files#diff-b3dee8ff07b1bb3b67f4ea9f3ca07dceR7) of `ErrorDisplay`. This really simplifies things when it's time to render because it means that `para` actually corresponds to paragraphs.

Horizon's `ED.para` conflates two commonly used but distinct things: "paragraphs" and space-separated text. It sounds like what you want is a shortcut for denoting sequences of space-separated text. What if I added a `spaced` list constructor to `error-display` that would denote a sequence of space-separated `ErrorDisplay` instances. A use would look something like:
```pyret
ED.paragraph([ED.spaced:
    ED.text("The"),
    ED.text("quick"),
    ED.highlight([ED.spaced: ED.text("brown"), ED.text("fox.")], someloc, somecolor)])
```